libsolidity/analysis/TypeChecker.cpp

```diff
@@ -2080,6 +2087,15 @@ void TypeChecker::typeCheckABIEncodeFunctions(
 			}
 		}
 
+		if (auto const inlineArray = dynamic_cast<InlineArrayType const*>(argType))
```

Comment:
Does this need to be a special case? Can't we just use the "mobile type"?

Reply:
Actually, it doesn't. We just need to check if the a mobile type of an index array is supported by the old ABI encoder. So that I will remove this code and update TypeChecker::typeSupportedByOldABIEncoder properly.